.gitignore

```diff
@@ -1,9 +1,42 @@
+
+# Created by https://www.toptal.com/developers/gitignore/api/rust,visualstudiocode
+# Edit at https://www.toptal.com/developers/gitignore?templates=rust,visualstudiocode
+
+### Rust ###
+# Generated by Cargo
+# will have compiled files and executables
+debug/
 target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
+
+# These are backup files generated by rustfmt
 **/*.rs.bk
 
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
+
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+!.vscode/*.code-snippets
```

Comment:
To me these exceptions suggest that these files are supposed to be tracked by git. Is that what you are implying here?

Reply:
Yes

Reply:
Why do you think these should be tracked?

Reply:
Well as the file says, I used [gitignore.io](https://www.toptal.com/developers/gitignore) which has been a lovel tool so far.

Indeed these should be tracked, as a vscode setup with tasks, launch options,... should be portable

Reply:
Okay, even firefox tracks some of these files. https://hg.mozilla.org/mozilla-central/file/tip/.vscode Though we have to be careful not to add any clutter or local settings.